src/dsp_tools/commands/ingest_xmlupload/upload_xml.py

```diff
@@ -0,0 +1,70 @@
+from __future__ import annotations
+
+from pathlib import Path
+
+from lxml import etree
+
+from dsp_tools.commands.ingest_xmlupload.apply_ingest_id import (
+    get_mapping_dict_from_file,
+    replace_filepath_with_sipi_id,
+)
+from dsp_tools.commands.xmlupload.upload_config import UploadConfig
+from dsp_tools.commands.xmlupload.xmlupload import xmlupload
+from dsp_tools.models.exceptions import InputError
+from dsp_tools.utils.create_logger import get_logger
+from dsp_tools.utils.xml_utils import remove_comments_from_element_tree
+
+logger = get_logger(__name__)
+
+
+def ingest_xmlupload(
+    xml_file: Path,
+    user: str,
+    password: str,
+    dsp_url: str,
+    sipi_url: str,
+) -> None:
+    """
+    This function reads an XML file
+    and imports the data described in it onto the DSP server,
+    using the ingest XML upload method.
+    Before using this method,
+    the multimedia files must be ingested on the DSP server.
+    A mapping file with the internal IDs of the multimedia files must also be provided.
+
+    Args:
+        xml_file: path to XML file containing the resources
+        user: the user's e-mail for login into DSP
+        password: the user's password for login into DSP
+        dsp_url: URL to the DSP server
+        sipi_url: URL to the Sipi server
+
+    Raises:
+        InputError: if any media was not uploaded or uploaded media was not referenced.
```

Comment:
I think we shouldn't prevent the xmlupload if there are files that were ingested but not in the XML. Imagine you ingested 100'000 files, and by mistake 1 was doubled. Now you want to make the xmlupload, but you are not allowed because there is 1 superfluous file on the server. That would be quite annoying...

Or is there a reason why we should disallow it?

Reply:
I see your point. But if a user gets this error, is there a way to clean it up? I imagine it to be difficult to access the server and then delete the unused file, and then delete it from the CSV.

Reply:
Do you mean a different error message?

Reply:
I think coping with bad user behaviour is something we only have to address once it is used widely. Apart from us, no one should use this route at the moment. To make it public we would need to add a start ingest command anyway, so I think the details of the implementation can wait.
As to more tolerance I strongly disagree, this code is the gate-keeper to bad user behaviour.
Uploading files that are never referenced and therefore float in nirvana may be user friendly but neither sustainable from a memory point of view nor good practice from a data modelling view. We had the discussion with Balduin about the circular references if we could not just have a requested number of IRIs. He declined that because this way the api allows is vulnerable to get data that is incorrect. This in my point of view is the same, user friendliness does not trump data integrity.

Reply:
Okay, that's a valid argument. So we know that we will have to work diligently and cleanly, if we ingest data.

Reply:
I think this is something that should not be decided by devs, that's a product decision. It could very well be a legit use case that projects ingest all their files at once, but create the resources in batches; but at the same time it is indeed a door-opener to bad user behaviour. Which one we pick is up to PM in the end.  
But surely this shouldn't keep this PR waiting, so I think for now, either way would be ok.